fdb-extensions/src/test/java/com/apple/foundationdb/async/hnsw/HNSWTest.java

```diff
@@ -379,6 +386,20 @@ private void validateSIFTSmall(@Nonnull final HNSW hnsw, final int k) throws IOE
                 int recallCount = 0;
                 for (final ResultEntry resultEntry : results) {
                     final int primaryKeyIndex = (int)resultEntry.getPrimaryKey().getLong(0);
+
+                    //
+                    // Assert that the original vector and the reconstructed vector are the same-ish vector
+                    // (minus reconstruction errors). The closeness value is dependent on the encoding quality settings,
+                    // the dimensionality, and the metric in use. For now, we just set it to 20.0 as that should be
```

Comment:
I suppose one could imagine asserting that the actual error from reconstructing the vector is within the error bounds of the estimator, if one wanted to

Reply:
That turns out it uses the same underlying math.